bigchaindb/consensus.py

```diff
@@ -1,244 +1,24 @@
-
-        # Check fulfillments
-        if not util.validate_fulfillments(transaction):
-            raise exceptions.InvalidSignature()
-
-        return transaction
+        """See `models.Transaction.validate` for documentation"""
+        return transaction.validate(bigchain)
 
     @staticmethod
     def validate_block(bigchain, block):
-        """Validate a block.
-
-        Args:
-            bigchain (Bigchain): an instantiated bigchaindb.Bigchain object.
-            block (dict): block to validate.
-
-        Returns:
-            The block if the block is valid else it raises an exception
-            describing the reason why the block is invalid.
-
-        Raises:
-            InvalidHash: if the hash of the block is wrong.
-        """
-
-        # Check if current hash is correct
-        calculated_hash = crypto.hash_data(util.serialize(block['block']))
-        if calculated_hash != block['id']:
-            raise exceptions.InvalidHash()
-
-        # Check if the block was created by a federation node
-        if block['block']['node_pubkey'] not in (bigchain.nodes_except_me + [bigchain.me]):
-            raise exceptions.OperationError('Only federation nodes can create blocks')
-
-        # Check if block signature is valid
-        verifying_key = crypto.VerifyingKey(block['block']['node_pubkey'])
-        if not verifying_key.verify(util.serialize(block['block']), block['signature']):
-            raise exceptions.InvalidSignature('Invalid block signature')
-
-        return block
-
-    @staticmethod
-    def create_transaction(owner_before, owner_after, tx_input, operation,
-                           payload=None):
-        """Create a new transaction
-
-        Refer to the documentation of ``bigchaindb.util.create_tx``
-        """
-
-        return util.create_tx(owner_before, owner_after, tx_input, operation,
-                              payload)
-
-    @staticmethod
-    def sign_transaction(transaction, private_key, bigchain=None):
-        """Sign a transaction
-
-        Refer to the documentation of ``bigchaindb.util.sign_tx``
-        """
-
-        return util.sign_tx(transaction, private_key, bigchain=bigchain)
+        """See `models.Block.validate` for documentation"""
+        return block.validate(bigchain)
 
     @staticmethod
-    def validate_fulfillments(signed_transaction):
-        """Validate the fulfillments of a transaction.
-
-        Refer to the documentation of ``bigchaindb.util.validate_fulfillments``
-        """
-
-        return util.validate_fulfillments(signed_transaction)
-
-    @staticmethod
-    def verify_vote_signature(block, signed_vote):
+    def verify_vote_signature(voters, signed_vote):
         """Verify the signature of a vote.
 
         Refer to the documentation of ``bigchaindb.util.verify_signature``
         """
-
-        return util.verify_vote_signature(block, signed_vote)
+        return verify_vote_signature(voters, signed_vote)
```

Comment:
As an aside: this method does nothing but passing the parameters to the util method of the same name. It most certainly will affect speed, as function calls take time.

This concern can be addressed in a separate issue as mentioned above.